tests/test_bootstrap.py

```diff
@@ -0,0 +1,551 @@
+import os
+
+import unittest
+
+try:
+    from unittest import mock
+except ImportError:
+    # `Python 2` or lower than `Python 3.3` does not
+    # have the `unittest.mock` module built-in
+    import mock
+from pythonforandroid.bootstrap import Bootstrap
+from pythonforandroid.distribution import Distribution
+from pythonforandroid.recipe import Recipe
+from pythonforandroid.archs import ArchARMv7_a
+from pythonforandroid.build import Context
+
+
+class BaseClassSetupBootstrap(object):
+    """
+    An class object which is intended to be used as a base class to configure
+    an inherited class of `unittest.TestCase`. This class will override the
+    `setUp` and `tearDown` methods.
+    """
+
+    def setUp(self):
+        self.ctx = Context()
+        self.ctx.ndk_api = 21
+        self.ctx.android_api = 27
+        self.ctx._sdk_dir = "/opt/android/android-sdk"
+        self.ctx._ndk_dir = "/opt/android/android-ndk"
+        self.ctx.setup_dirs(os.getcwd())
+        self.ctx.recipe_build_order = [
+            "hostpython3",
+            "python3",
+            "sdl2",
+            "kivy",
+        ]
+
+    def setUp_distribution_with_bootstrap(self, bs):
+        """
+        Extend the setUp by configuring a distribution, because some test
+        needs a distribution to be set to be properly tested
+        """
+        self.ctx.bootstrap = bs
+        self.ctx.bootstrap.distribution = Distribution.get_distribution(
+            self.ctx, name="test_prj", recipes=["python3", "kivy"]
+        )
+
+    def tearDown(self):
+        """
+        Extend the `tearDown` by configuring a distribution, because some test
+        needs a distribution to be set to be properly tested
+
+        .. note:: Here we need to reset `self.ctx.bootstrap.distribution`
+                  because, otherwise we will be unable to successfully pass the
+                  test :meth:`TestBootstrapBasic.test_attributes`
+        """
+        if self.ctx.bootstrap:
+            self.ctx.bootstrap.distribution = None
+        self.ctx.bootstrap = None
+
+
+class TestBootstrapBasic(BaseClassSetupBootstrap, unittest.TestCase):
```

Comment:
I'm already not a big fan of multiple inheritance, even for mixins. And that's my personal opinion.
But here anyway `BaseClassSetupBootstrap` was already inheriting `unittest.TestCase` what do you think?

Reply:
Why you don't like `multiple-inheritance`?
In some situations it makes a lot of sense to use it...it's because of the `MRO` which could be tricky to follow?

Reply:
Yes mainly because it can get things harder to follow sometimes. I've learn programming the OOP way and I think I tend to abuse inheritance in general.
Anyway I think your code is pretty straight forward and readable